### PR TITLE
fast-pin: Cortex M3 Bit-Band access to perform fast GPIO atomic read-modify-write

### PR DESCRIPTION
**Pull request by satishgn**
The bit-banding code is common across all STM32Fx series

### CODE REVIEW

File: hal/src/core/pinmap_impl.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ extern "C" {
 typedef struct STM32_Pin_Info {
   GPIO_TypeDef* gpio_peripheral;
   pin_t gpio_pin;
+  uint8_t gpio_pin_source;
```
Comment (m-mcgowan) [inline hal/src/core/pinmap_impl.h]: This must come at the end of the pinmap or older applications using the old version of pinmap will crash if newer system firmware is updated.

**Comment by pkourany** [comment]
Bad news - doing a straight sequence of pinSetFast() and pinResetFast() with bitbanding gives a pulse time of 126ns.  Without bitbanding (straight to register), the same sequences gives a pulse time of 42ns.  So bitbanding is much slower!

File: hal/src/core/pinmap_impl.h (hunk 1 above)
Comment (satishgn) [inline hal/src/core/pinmap_impl.h]: The above addition was for the Core...

Comment (m-mcgowan) [inline hal/src/core/pinmap_impl.h]: Totally, I only noticed afterwards. I should have written a follow up note -  my appologies for the noise!

**Pull request merged**